src/CPUBackend.jl

```diff
@@ -9,18 +9,18 @@ struct CPUEvent end
 wait(ev::CPUEvent) = nothing
 wait(evs::AbstractArray{CPUEvent}) = wait.(evs)
 
-@inline function split(ndrange::CartesianIndices, nthreads::NTuple{N,T}) where {N,T}
-    nblocks = ntuple(Val(N)) do I
+@inline function split(ndrange::CartesianIndices, nthreads::NTuple)
+    nblocks = ntuple(Val(length(nthreads))) do I
         return cld(size(ndrange, I), nthreads[I])
     end
     return nblocks
 end
 
-function (k::Kernel{<:CPUDevice})(args::Vararg{Any,B}; ndrange::NTuple{N,T}, priority=:low, nthreads=nothing) where {N,T,B}
+function (k::Kernel{<:CPUDevice})(args::Vararg{Any,B}; ndrange, priority=:low, nthreads=nothing) where {B}
```

Comment:
Revert `::NTuple` annotation to allow for more general type of `ndrange`. 
@albert-de-montserrat feel free to check  if it's fine wrt type stability you were investigating.
@utkinis I was checking and it seems that prior to type stability fixes it did also fail if `ndrange=CartesianIndices(A)` in function call on `ndrange_to_indices()`

Reply:
@luraess Thanks for checking, added the missing method!

Reply:
Looks good to me. It's able to do constant propagation on `ndims(::CartesianIndices)`
```julia-repl
julia> x
CartesianIndices((4:29, 4:29))

julia> @code_warntype ndims(x)
MethodInstance for ndims(::CartesianIndices{2, Tuple{UnitRange{Int64}, UnitRange{Int64}}})
  from ndims(R::CartesianIndices) @ Base.IteratorsMD multidimensional.jl:378
Arguments
  #self#::Core.Const(ndims)
  R::CartesianIndices{2, Tuple{UnitRange{Int64}, UnitRange{Int64}}}
Body::Int64
1 ─ %1 = Base.IteratorsMD.typeof(R)::Core.Const(CartesianIndices{2, Tuple{UnitRange{Int64}, UnitRange{Int64}}})
│   %2 = Base.IteratorsMD.ndims(%1)::Core.Const(2)
└──      return %2
```